File: rounds/943/prefiquence/main.ts
import { type TestCase, run } from "../../utils";

export class Prefiquence implements TestCase {
  a: string;
  b: string;

  constructor(lines: Array<string>) {
    const [x, y] = lines.shift()!.split(' ').map(Number);
    this.a = lines.shift()!.slice(0, x);
    this.b = lines.shift()!.slice(0, y);
  }

  run(): number {
    return prefiquence(this.a, this.b);
  }
}

export function prefiquence(a: string, b: string): number {
  let k = 0;
  for (let j = 0; k < a.length && j < b.length; ++j) {
    if (a[k] == b[j]) {
      ++k;
    }
  }
  return k;
}

run(Prefiquence);

File: rounds/943/prefiquence/main.cc
#include "utils.h"

void run(istream &in, ostream &out, int t)
{
  int x, y;
  string a, b;
  in >> x >> y >> a >> b;
  const auto f = [&a, &b]()
  {
    int k = 0;
    for (int j = 0; k < a.size() && j < b.size(); ++j)
    {
      if (a[k] == b[j])
        ++k;
    }
    return k;
  };
  out << f() << endl;
}

int main()
{
  INPUT(__FILE__);
  int c;
  input >> c;
  for (int i = 1; i <= c; ++i)
    run(input, cout, i);
}